### PR TITLE
issue-81: is readFully necessary for Input interface

### PR DESCRIPTION
**Pull request by colinmjj**
The pull request for https://github.com/intel-hadoop/chimera/issues/81

### CODE REVIEW

File: src/main/java/com/intel/chimera/stream/PositionedCryptoInputStream.java [hunk 1]
```diff
@@ -120,13 +121,29 @@ public int read(long position, byte[] buffer, int offset, int length)
...
     checkStream();
-    input.readFully(position, buffer, offset, length);
+    readFullyInternal(position, buffer, offset, length);
     if (length > 0) {
       // This operation does not change the current offset of the file
       decrypt(position, buffer, offset, length);
     }
   }
 
+  /**
```
Comment (jerrychenhf) [inline src/main/java/com/intel/chimera/stream/PositionedCryptoInputStream.java]: Can this put to Utils as static method?

File: src/test/java/com/intel/chimera/stream/PositionedCryptoInputStreamTest.java [hunk 2]
```diff
@@ -359,19 +359,6 @@ public int read(long position, byte[] buffer, int offset, int length)
...
-        int nbytes = read(position+nread, buffer, offset+nread, length-nread);
-        if (nbytes < 0) {
-          throw new EOFException("End of file reached before reading fully.");
-        }
-        nread += nbytes;
-      }
-    }
-
-    @Override
     public void seek(long position) throws IOException {
```
Comment (sundapeng) [inline src/test/java/com/intel/chimera/stream/PositionedCryptoInputStreamTest.java]: It's better to add comment for public method.

File: src/main/java/com/intel/chimera/input/StreamInput.java [hunk 3]
```diff
@@ -74,13 +74,6 @@ public int read(long position, byte[] buffer, int offset, int length)
...
 
   @Override
-  public void readFully(long position, byte[] buffer, int offset, int length)
-      throws IOException {
-    throw new UnsupportedOperationException(
-        "Positioned readFully is not supported by this implementation");
-  }
-
-  @Override
   public void seek(long pos) throws IOException {
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/input/StreamInput.java]: It's better to add comment for public method.

File: src/main/java/com/intel/chimera/input/ChannelInput.java [hunk 4]
```diff
@@ -78,13 +78,6 @@ public int read(long position, byte[] buffer, int offset, int length)
...
 
   @Override
-  public void readFully(long position, byte[] buffer, int offset, int length)
-      throws IOException {
-    throw new UnsupportedOperationException(
-        "Positioned readFully is not supported by this implementation");
-  }
-
-  @Override
   public void seek(long pos) throws IOException {
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/input/ChannelInput.java]: It's better to add comment for public method.

File: src/test/java/com/intel/chimera/stream/PositionedCryptoInputStreamTest.java (hunk 2 above)
Comment (colinmjj) [inline src/test/java/com/intel/chimera/stream/PositionedCryptoInputStreamTest.java]: The comment for the public method is added in the interface Input.

File: src/main/java/com/intel/chimera/stream/PositionedCryptoInputStream.java (hunk 1 above)
Comment (colinmjj) [inline src/main/java/com/intel/chimera/stream/PositionedCryptoInputStream.java]: I'll move the method to IOUtils.

**Comment by jerrychenhf** [comment]
LGTM. I noticed that you use the same commit message for all the commits in the pull request. Which will leads multiple commits in the master with the same messages which is not the right way.  Please take this note for consideration in your future commits.

**Pull request merged**